File: Baekjoon/2217.cpp
#include <iostream>
#include <vector>
#include <algorithm>
using namespace std;

int main() {
	int n;
	cin >> n;
	vector<int> v;
	for (int i = 0; i < n; i++) {
		int tmp;
		cin >> tmp;
		v.push_back(tmp);
	}
	sort(v.begin(), v.end());
	int ans = 0;
	for (int i = 0; i < n; i++) {
		ans = max(ans, v[i] * (n - i));
	}
	cout << ans << '\n';	
}

File: Baekjoon/2217.kt
import kotlin.math.max

fun main() {
    val n = readLine()!!.toInt()
    val r = IntArray(n) { 0 }
    for (i in 0 until n) r[i] = readLine()!!.toInt()
    r.sort()
    var ans = 0
    for (i in 0 until n) ans = max(ans, r[i] * (n - i))
    println(ans)
}